automatic/sublimetext4/tools/chocolateyInstall.ps1

```diff
@@ -8,7 +10,23 @@ $packageArgs = @{
   checksumType           = 'sha256'
   silentArgs             = '/VERYSILENT /NORESTART /TASKS="contextentry"'
   validExitCodes         = @(0)
-  softwareName           = 'Sublime Text 4'
+  softwareName           = $softwareName
 }
 
 Install-ChocolateyPackage @packageArgs
+
+[array]$key = Get-UninstallRegistryKey -SoftwareName $softwareName
+
+if ($key.Count -eq 1) {
+  $key | ForEach-Object {
+    $sublInstallLocation = Join-Path -Path $_.InstallLocation -ChildPath 'subl.exe'
+    Install-BinFile -Name 'subl' -Path $sublInstallLocation
+  }
+} elseif ($key.Count -eq 0) {
+  Write-Warning "Not installing subl.exe shim, unable to determine install location"
+} elseif ($key.Count -gt 1) {
+  Write-Warning "$($key.Count) matches found!"
+  Write-Warning "Not installing subl.exe shim, too many install locations found"
+  Write-Warning "Please alert package maintainer the following keys were matched:"
+  $key | ForEach-Object {Write-Warning "- $($_.InstallLocation)"}
+}
```

Comment:
Per other review comments, this code will probably need to be used by both `chocolateyInstall.ps1` and `chocolateyUninstall.ps1` so it would be great to split it into a separate function in a `helpers.ps1` file.

Reply:
The behaviours of the `ForEach-Object` code block differs, as do the `Write-Warning` strings, depending on install vs. uninstall context.  I don't know how to code a single helper function to handle both scenarios.  A big problem is I'm not a PowerShell expert (I live more in the C and Python world, and in UNIX space, not so much Windows).  :)

Reply:
That's fine, if you do it separately I'll play with it myself and either figure it out or give up and leave it separate :) I'm not powershell expert either so we'll see.